File: Pelikula.API/Model/Prodaja/ProdajaResponse.cs
using Pelikula.API.Model.Rezervacija;
using System;
using System.Collections.Generic;

namespace Pelikula.API.Model.Prodaja
{
    public class ProdajaResponse
    {
        public int Id { get; set; }
        public string BrojRacuna { get; set; }
        public DateTime Datum { get; set; }

        public decimal UkupnaCijena { get; set; }

        public LoV Korisnik { get; set; }
        public ICollection<ProdajaArtikalResponse> ProdajaArtikal { get; set; }
        public RezervacijaResponse Rezervacija { get; set; }

        public bool jeIskoristeno { get; set; }

        public decimal GetUkupnaCijena(ICollection<ProdajaArtikalResponse> prodajaArtikli, RezervacijaResponse rezervacija) {
            decimal ukupnaCijena = 0;

            foreach (var prodajaArtikal in prodajaArtikli)
                ukupnaCijena += (prodajaArtikal.Artikal.Cijena * prodajaArtikal.Kolicina);

            if (rezervacija != null)
                ukupnaCijena += rezervacija.Cijena;

            return Math.Round(ukupnaCijena, 2);
        }

    }
}

File: Pelikula.MOBILE/lib/model/prodaja/prodaja_response.dart
import 'package:pelikula_mobile/model/lov.dart';
import 'package:pelikula_mobile/model/rezervacija/rezervacija_response.dart';

class ProdajaResponse {
  int? id;
  String? brojRacuna;
  DateTime? datum;

  double? ukupnaCijena;

  LoV? korisnik;
  RezervacijaResponse? rezervacija;

  List<dynamic>? prodajaArtikal;

  ProdajaResponse({
    this.id,
    this.brojRacuna,
    this.datum,
    this.ukupnaCijena,
    this.korisnik,
    this.rezervacija,
    this.prodajaArtikal,
  });

  factory ProdajaResponse.fromJson(Map<String, dynamic> json) {
    return ProdajaResponse(
      id: json['id'] as int,
      brojRacuna: json['brojRacuna'] as String,
      datum: DateTime.tryParse(json['datum']),
      ukupnaCijena: json['ukupnaCijena'] as double,
      korisnik:
          json['korisnik'] != null ? LoV.fromJson(json['korisnik']) : null,
      rezervacija: json['rezervacija'] != null
          ? RezervacijaResponse.fromJson(json['rezervacija'])
          : null,
      prodajaArtikal: json['prodajaArtikal'] != null
          ? json['prodajaArtikal'] as List
          : null,
    );
  }

  Map<String, dynamic> toJson() => {
        "id": id,
        "brojRacuna": brojRacuna,
        "datum": datum == null ? null : datum!.toIso8601String(),
        "ukupnaCijena": ukupnaCijena,
        "korisnik": korisnik == null ? null : korisnik,
        "rezervacija": rezervacija,
        "prodajaArtikal": prodajaArtikal,
      };
}
